src/webgl/material.js

```diff
@@ -718,171 +725,521 @@ p5.prototype.createFilterShader = function (fragSrc) {
  * @example
  * <div modernizr='webgl'>
  * <code>
- * // Note: A "uniform" is a global variable within a shader program.
+ * // Example 1: Basic Fill Shader without Lights
+ * // Applies a custom shader to the fill of objects
+ * let fillShader;
  *
- * // Create a string with the vertex shader program.
- * // The vertex shader is called for each vertex.
  * let vertSrc = `
  * precision highp float;
- * uniform mat4 uModelViewMatrix;
- * uniform mat4 uProjectionMatrix;
- *
  * attribute vec3 aPosition;
- * attribute vec2 aTexCoord;
- * varying vec2 vTexCoord;
- *
  * void main() {
- *   vTexCoord = aTexCoord;
- *   vec4 positionVec4 = vec4(aPosition, 1.0);
- *   gl_Position = uProjectionMatrix * uModelViewMatrix * positionVec4;
+ *   gl_Position = vec4(aPosition, 1.0);
  * }
  * `;
  *
- * // Create a string with the fragment shader program.
- * // The fragment shader is called for each pixel.
  * let fragSrc = `
  * precision highp float;
- *
+ * uniform float uMouseX;
  * void main() {
- *   // Set each pixel's RGBA value to yellow.
- *   gl_FragColor = vec4(1.0, 1.0, 0.0, 1.0);
+ *   vec3 color = vec3(uMouseX, 0.0, 1.0); // Color based on mouse X
+ *   gl_FragColor = vec4(color, 1.0);
  * }
  * `;
  *
  * function setup() {
  *   createCanvas(100, 100, WEBGL);
+ *   fillShader = createShader(vertSrc, fragSrc);
+ *   shader(fillShader);
+ *   noStroke();
+ *   describe('A square with color changing based on mouse X without lighting.');
+ * }
  *
- *   // Create a p5.Shader object.
- *   let shaderProgram = createShader(vertSrc, fragSrc);
+ * function draw() {
+ *   fillShader.setUniform('uMouseX', map(mouseX, 0, width, 0, 1));
+ *   plane(100, 100);
+ * }
+ * </code>
+ * </div>
  *
- *   // Apply the p5.Shader object.
- *   shader(shaderProgram);
+ * @example
+ * <div modernizr='webgl'>
+ * <code>
+ * // Example 2: Fill Shader with Lights Affected by Mouse
```

Comment:
okay, when I say `lights()` I don't mean to say lightPos or something you have made
```
let lightPos = [(mouseX - width / 2) / width, (mouseY - height / 2) / height, 1.0];
 *   fillShader.setUniform('uLightPos', lightPos); // Mouse changes light position
 ```
 Since `shader()` function was not running when we had `lights()` called in our sketch, we need to have certian uniforms or attributes to make that run. So you can use lights() function instead of that.

Reply:
Since this example is mostly the same than the above, you may use lights() in the above example and with a comment that lights() getting ignored as shader is controlling the fill part. For this one, maybe we could have a sketch where we can see `texture` is being effected by shader().